---
title: "RNA-Seq Analysis"
author: "Jason Hunter and Ryan Greer"
date: "`r Sys.Date()`"
output: 
  html_document:
    keep_md: yes
    toc: true
    toc_float: true
    theme: cerulean
    highlight: tango
    code_folding: hide
    df_print: kable
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Installing Packages
```{r install-packages, eval=FALSE}
# every single install.packages() command we ran on fiji (may not be exhaustive)
# NOTE: This chunk is set to eval=FALSE.
# Run these lines interactively in your R console *only* if you need to install these packages.
# If running on a system where packages are already installed, you can ignore this.
# options(repos = c(CRAN = "https://cloud.r-project.org")) # may need to be commented out if not compliling locally
options(repos = c(CRAN = "https://cloud.r-project.org")) # may need to be commented out if not compliling locally
install.packages(c("tidyverse",
                   "pheatmap",
                   "textshape",
                   "Rcpp",
                   "magrittr",
                   "ggplot2",
                   "dplyr",
                   "IRanges",
                   "purrr",
                   "readr",
                   "tibble",
                   "tidyr",
                   "matrixStats",
                   "broom",
                   "reshape",
                   "reshape2",
                   "igraph",
                   "corrplot",
                   "DT"))
# Install BiocManager
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.20") # specify version if needed
BiocManager::install(c("DESeq2", "apeglm"))
```

## Loading Required Libraries
```{r load-libraries, eval=TRUE, message=FALSE, warning=FALSE}
# loading in every library we used over the semester
library(tidyverse)
library(DESeq2)
library(magrittr)
library(ggplot2)
library(IRanges)
library(pheatmap)
library(textshape)
library(Rcpp)
library(matrixStats)
library(broom)
library(igraph)
library(corrplot)
library(DT)
```

## Methods Summary

RNA-Sequencing data was processed using the nf-core/rnaseq pipeline, with quantification performed by Salmon. Initial differential expression analysis was conducted using DESeq2.

**For a comprehensive description of the data processing pipeline, quality control, normalization, and DESeq2 setup, please see the [Methods Appendix](knit_me_A.html).**

## Introduction

This analysis explores RNA-Seq data from mouse embryonic stem cells (mESCs) after doxycycline exposure over a time course (0, 12, 24, 48, 96 hours). The goals were to identify significantly differentially expressed genes, perform co-expression network analyses, and uncover biological modules relevant to mitochondrial function, inflammation, differentiation, and metabolic shifts.

## Importing Counts and TPM Values as well as the Significantly Changed Genes
```{r import-data-on-most-signicantly-changed-genes}
load("DESeqResults/DESEQ_results.rdata")
load("DESeqResults/TPM_results.rdata")

# loading in the genes that significantly changed (from the DESeq2 analysis)
data_sig_4fold  <- read.table("DESeqResults/sig_4fold_genes_counts.tsv",
                              header = TRUE,
                              sep = "\t")

gene_names <- read.csv("DESeqResults/gene_names.csv",
                       header = TRUE,
                       stringsAsFactors = FALSE)

counts     <- read.table("DESeqResults/salmon.merged.gene_counts.tsv",
                         header = TRUE,
                         sep = "\t",
                         stringsAsFactors = FALSE)

tpms       <- read.table("DESeqResults/salmon.merged.gene_tpm.tsv",
                         header = TRUE,
                         sep = "\t",
                         stringsAsFactors = FALSE)
```

## First we created a volcano plot to get a good visual representation of how the genes are distributed.
```{r volcano-plot, fig.cap="Volcano plot showing differentially expressed genes based on DESeq2 results. Blue points are significantly upregulated (padj < 0.05, log2FC > 1), red points are significantly downregulated (padj < 0.05, log2FC < -1).", fig.width=7, fig.height=5}
# volcano plot from 'filtered_res_df',
# which is a data frame we created
# from the results/DESEQ_results.rdata

# adjust thresholds how we see fit
# the max p-value we want to see
# and the min log2fc we want to see
padj_cutoff <- 0.05
log2fc_cutoff <- 1

# add simple factor columns for coloring:
# creates a new column by mutate()
# called 'sig_flag' in filtered_res_df
# and assigns a value based on the conditions
# using case_when() to determine whether a gene is:
# upregulated, downregulated, or not significant
sig_flag_filtered_res_df <- filtered_res_df %>%
  mutate(
    sig_flag = case_when(
      (padj < padj_cutoff & log2FoldChange >  log2fc_cutoff) ~ "Up",
      (padj < padj_cutoff & log2FoldChange < -log2fc_cutoff) ~ "Down",
      TRUE ~ "NotSig"
    )
  )

volcano_plot <- ggplot(sig_flag_filtered_res_df,
       aes(x = log2FoldChange,
           y = -log10(padj),
           color = sig_flag)) +
  geom_point(alpha = 0.7) +
  scale_color_manual(values = c("Up" = "blue",
                                "Down" = "red",
                                "NotSig" = "grey60")) +
  geom_vline(xintercept = c(-log2fc_cutoff, log2fc_cutoff),
             linetype = "dashed") +
  geom_hline(yintercept = -log10(padj_cutoff),
             linetype = "dashed") +
  labs(
    title = "Volcano Plot of Differential Expression Results",
    subtitle = "Doxycycline Treatment",
    x = "Log2 Fold Change",
    y = "-Log10(Adjusted p-value)"
  ) +
  theme_minimal()
# save the image in the figures folder
ggsave(filename = "figures/Graphs/volcano_plot.png",
       plot = volcano_plot,
       width = 6,
       height = 4)
```

```{r volcano-plot-figure, echo=FALSE}
knitr::include_graphics("figures/Graphs/volcano_plot.png")
```


## There's a ton of activity amongst genes in the volcano plot, both upregulated and downregulated.
## Lets take a look at just the dataframe of genes that are P < 0.01 & that change greater that 4 fold (up or down)
## We calculated this in 06_Differential_expression_analyses/04_exploring_results.Rmd
## We can see that the genes that significantly changed are:
<div style="column-count: 3;">
```{r print-significant-genes, echo=FALSE, results='asis'}
markdown_list <- paste("* ", data_sig_4fold$gene_name, collapse = "\n")
cat(markdown_list)
```
</div>

## There's a lot of 'Gm' genes in this list, as well as some predicted genes with some funky names.
## Let's filter them since they are likely not of interest.
```{r clean-&-sort-gene-data}
data <- data_sig_4fold[!grepl("Gm", data_sig_4fold$gene_name), ]
data_cleaned <- data[!grepl("Rik", data$gene_name), ]
# lets sort it too (alphabetically), why not
data_cleaned <- data_cleaned[order(data_cleaned$gene_name,
                                   decreasing = FALSE), ]
# there were also a couple of genes that were duplicates,
# so we'll remove them as well
data_cleaned <- data_cleaned[!duplicated(data_cleaned$gene_name), ]
```
<div style="column-count: 3;">
```{r echo=FALSE, results='asis'}
markdown_list <- paste("* ", data_cleaned$gene_name, collapse = "\n")
cat(markdown_list)
```
</div>
```{r prepare-gene-timecourse-data}
# we can also make a list of all the genes we filtered as individual dataframes
# this will make it easier to work with them in my opinion
gene_data_list <- lapply(data_cleaned$gene_name, function(gene) {
  data_cleaned[data_cleaned$gene_name == gene, ]
})
names(gene_data_list) <- data_cleaned$gene_name

## now we reshape data for time course analysis by melting each gene dataframe
gene_long_list <- lapply(gene_data_list, function(df) {
  df %>% pivot_longer(cols = -gene_name,
                      names_to = "sample",
                      values_to = "count")
})

## now we can extract the time point and replicate number
## from the sample column for each gene
gene_long_list <- lapply(gene_long_list, function(df) {
  df$timepoint <- gsub("WT_([0-9]+)_[0-9]+", "\\1", df$sample)
  df$replicate <- gsub("WT_[0-9]+_([0-9]+)", "\\1", df$sample)
  df$timepoint <- factor(df$timepoint, levels = c("0", "12", "24", "48", "96"))
  df
})
```


## Calculating the mean and standard error for each time point
```{r mean-and-se}
## list of dataframes with summary statistics for each gene
gene_summary_list <- lapply(gene_long_list, function(df) {
  df %>%
    group_by(timepoint) %>%
    summarise(
      mean = mean(count),
      se = sd(count) / sqrt(n()),
      sd = sd(count),
      .groups = "drop"
    )
})
```

## Plotting the mean + standard error of each gene for each time point as a facet plot
```{r plot-gene-timecourse-data, fig.cap="Mean expression (+/- Standard Error) over time for significantly differentially expressed genes (after filtering). Each panel represents one gene.", fig.width=12, fig.height=8, message=FALSE, warning=FALSE}
for (gene in names(gene_summary_list)) {
  df <- gene_summary_list[[gene]]
  df$timepoint <- as.numeric(as.character(df$timepoint)) 
  p <- ggplot(df, aes(x = timepoint, y = mean, group = 1)) +
    geom_line() +
    geom_point() +
    geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
    scale_x_continuous(breaks = unique(df$timepoint)) +
    labs(
      title = paste(gene, "Expression Across Time"),
      y = "Mean Count",
      x = "Time (hours)",
      caption = "Error bars represent standard error of the mean"
    ) +
    theme(
      plot.title = element_text(hjust = 0.5, face = "bold"),
      axis.title = element_text(face = "bold")
    )

# save the image in the figures folder
ggsave(filename = paste0("figures/TPM_plots/", gene, "_expression.png"), plot = p, width = 6, height = 4)
}

# combine all gene summaries with an added gene column
all_summary <- dplyr::bind_rows(gene_summary_list, .id = "gene")
# theres some plots with a lot of standard deviation

# this facet plot will show all the genes in a single .png file
all_summary$timepoint <- as.numeric(as.character(all_summary$timepoint)) 
facet_plot <- ggplot(all_summary, aes(x = as.numeric(timepoint), y = mean, group = gene)) +
  geom_line() +
  geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(all_summary$timepoint)) +
  facet_wrap(~ gene, scales = "free_y") +
  labs(
    title = "Expression Across Time",
    y = "Mean Count",
    x = "Time (hours)",
  ) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold"),
    axis.title = element_text(face = "bold")
  )

# save the facet plot to a file
ggsave(filename = "figures/TPM_plots/all_genes_facet_expression.png",
       plot = facet_plot,
       width = 12,
       height = 8)
print(facet_plot)
```

## We also conducted a statistical analysis of expression changes where we compare each time point to the 0 hour time point
```{r comparing-to-the-zero-hour-timepoint}
# perform t-tests for each gene at each time point
timepoints <- c("12", "24", "48", "96")
# create a list to store results
stat_results_list <- lapply(names(gene_long_list), function(gene) {
  df <- gene_long_list[[gene]]
  gene_stats <- data.frame()
  # loop through each time point
  for (tp in timepoints) {
    # filter data for the current gene and timepoint
    tp_data <- df %>% filter(timepoint %in% c("0", tp))
    # proceed only if there is data for both timepoints
    if (nrow(tp_data %>% filter(timepoint == "0")) > 0 &&
          nrow(tp_data %>% filter(timepoint == tp)) > 0) {
      # perform t-test
      t_test <- t.test(count ~ timepoint, data = tp_data)
      # calculate mean for each timepoint
      mean_tp <- mean(tp_data$count[tp_data$timepoint == tp])
      # calculate mean for timepoint 0
      mean_0 <- mean(tp_data$count[tp_data$timepoint == "0"])
      # calculate fold change
      fc <- mean_tp / mean_0
      # store results in a data frame
      gene_stats <- rbind(gene_stats,
                          data.frame(gene = gene,
                                     comparison = paste0("0 vs ", tp),
                                     p_value = t_test$p.value,
                                     fold_change = fc))
    }
  }
  gene_stats
})

# combining all results into one data frame:
stat_results_all <- do.call(rbind, stat_results_list)
DT::datatable(stat_results_all,
              options = list(pageLength = 10),
              caption = "T-test results comparing each time point to 0 hours for significantly changed genes.")
```

## Heatmap Visualization
```{r heatmap-visualization}

# create a heatmap matrix
heatmap_matrix <- data_cleaned %>%
  distinct(gene_name, .keep_all = TRUE) %>%
  select(gene_name, starts_with("WT")) %>%
  column_to_rownames("gene_name") %>%
  mutate(across(everything(), as.numeric)) %>%
  replace(is.na(.), 0) %>%  # replace NAs with zeros
  as.matrix()

# check for infinite or NaN values explicitly
if (any(is.infinite(heatmap_matrix) | is.na(heatmap_matrix))) {
  heatmap_matrix[!is.finite(heatmap_matrix)] <- 0
}

# log2 transformation
heatmap_matrix_log <- log2(heatmap_matrix + 1)

# generate heatmap
pheatmap(heatmap_matrix_log,
         scale = "row",
         clustering_distance_rows = "correlation",
         fontsize_row = 8,
         main = "Log2 Expression Heatmap")
```

## Co-expression Network Analysis
```{r network-analysis, warning = FALSE}
# calculate correlation matrix
cor_matrix <- cor(t(heatmap_matrix), method = "pearson")

# define threshold correlations
threshold <- 0.7
network_matrix <- cor_matrix
network_matrix[abs(network_matrix) < threshold] <- 0
diag(network_matrix) <- 0

# build network
network <- graph_from_adjacency_matrix(network_matrix,
                                       weighted = TRUE,
                                       mode = "undirected")

# community detection
communities <- cluster_walktrap(network, weights = abs(E(network)$weight))
V(network)$module <- communities$membership
V(network)$color <- rainbow(max(V(network)$module))[V(network)$module]

# plot network
plot(network,
     vertex.size = 10,
     vertex.label.cex = 0.8,
     vertex.label.color = "black",
     edge.width = abs(E(network)$weight)*2,
     edge.color = ifelse(E(network)$weight > 0, "blue", "red"),
     main = "Gene Co-expression Network")

legend("topright", legend = paste("Module", 1:max(V(network)$module)),
       col = rainbow(max(V(network)$module)), pch = 19, bty = "n")
```

## Key Gene Expression Changes

Here we highlight specific genes showing interesting expression patterns and potential biological relevance.

## AOC3 Downregulation
We identified significant downregulation of a lesser-known gene that contributes to inflammation, through the production of the oxidative VAP-1(vascular adhesion protein). 
This protein is thought to contribute to the progression of vascular disorders and kidney complications. 
Additionally, its levels have been shown to be correlated with all-cause mortality rates in type 2 diabetics. 
We observed a ~6.25-fold reduction of AOC3 expression from the 0 to 96 hours timepoints. 
Whether this trend continues further from doxycycline exposure or is a more short-term change in expression remains to be seen.

Li HY, Jiang YD, Chang TJ, Wei JN, Lin MS, Lin CH, Chiang FT, Shih SR, Hung CS, Hua CH, Smith DJ, Vanio J, Chuang LM. Serum vascular adhesion protein-1 predicts 10-year cardiovascular and cancer mortality in individuals with type 2 diabetes. Diabetes. 2011 Mar;60(3):993-9. doi: 10.2337/db10-0607. Epub 2011 Jan 31. PMID: 21282368; PMCID: PMC3046860.

```{r plot-aoc3, echo=FALSE}
# Find the specific summary data for Aoc3
aoc3_summary <- gene_summary_list[["Aoc3"]]
aoc3_summary$timepoint <- as.numeric(as.character(aoc3_summary$timepoint))

ggplot(aoc3_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(aoc3_summary$timepoint)) +
  labs(title = "Aoc3 Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## AOC3 IGV
![alt text](figures/IGV_pics/AOC3_IGV.png)

```{r AOC3}
aoc3_results <- stat_results_all %>% filter(gene == "Aoc3")
knitr::kable(aoc3_results, digits = 3)
```

## KLF17 Downregulation
We saw that the expression of the KLF17 gene had a sharp reduction in expression, reaching its lowest at 48 hours(~5.75 fold reduction). 
Interestingly, the expression seemed to rebound over the next 48 hours, so we aren't sure if this trend will continue, with expression returning to baseline. 
KLF17 (Krueppel-like factor 17), is a transcription factor that is involved in the processes of stem cell differentiation, so reduced levels may influence the direction that these cells take, though we are not sure in what capacity.

```{r plot-klf17, echo=FALSE}
# Find the specific summary data for Klf17
klf17_summary <- gene_summary_list[["Klf17"]]
klf17_summary$timepoint <- as.numeric(as.character(klf17_summary$timepoint))

ggplot(klf17_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(klf17_summary$timepoint)) +
  labs(title = "Klf17 Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## KLF17 IGV
![KLF17_IGV](figures/IGV_pics/klf17_IGV.png)

```{r KLF17}
klf17_results <- stat_results_all %>% filter(gene == "Klf17")
knitr::kable(klf17_results, digits = 3)
```

## ABCC2 Downregulation
ABCC2 (ATP-binding cassette subfamily C member 2) is a gene associated with drug resistance, and plays a role in optimal functioning of the kidney and liver.
It is involved in the transportation of foreign substances, toxins, and drugs inside the body.
With this role, the significant, 4-fold reduction we see in ABCC2 expression after exposure to doxycycline may lead to increased inflammation, since ABCC2 is involved in the transportation of the toxic and inflammatory substances out of the system.

```{r plot-abcc2, echo=FALSE}
# Find the specific summary data for Abcc2
abcc2_summary <- gene_summary_list[["Abcc2"]]
abcc2_summary$timepoint <- as.numeric(as.character(abcc2_summary$timepoint))

ggplot(abcc2_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(abcc2_summary$timepoint)) +
  labs(title = "Abcc2 Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## ABCC2 IGV
![ABCC2_IGV](figures/IGV_pics/Abcc2_IGV.png)

```{r Abcc2}
abcc2_results <- stat_results_all %>% filter(gene == "Abcc2")
knitr::kable(abcc2_results, digits = 3)
```

## NLRP3 Downregulation
NLRP3 (NLR family pyrin domain containing 3) is a protein heavily involved in regulating immune inflammatory response, through its role in the inflammasome, a protein complex involved in the detection of cell damage and stress. 
Since NLRP3 activates inflammatory signaling, lower levels are likely to contribute to the inflammation-lowering properties that doxycycline is famous for.
We observed a ~4.5-fold reduction in NLRP3 levels from the 0 to 96 hour timepoints, so we believe this gene is one of the key players in doxycycline's effects on stress/inflammatory response.

```{r plot-nlrp3, echo=FALSE}
# Find the specific summary data for Nlrp3
nlrp3_summary <- gene_summary_list[["Nlrp3"]]
nlrp3_summary$timepoint <- as.numeric(as.character(nlrp3_summary$timepoint))

ggplot(nlrp3_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(nlrp3_summary$timepoint)) +
  labs(title = "Nlrp3 Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## NLRP3 IGV
![NLRP3_IGV](figures/IGV_pics/Nlrp3_IGV.png)
```{r NLRP3}
nlrp3_results <- stat_results_all %>% filter(gene == "Nlrp3")
knitr::kable(nlrp3_results, digits = 3)
```

## KNG1 Downregulation:
KNG1 (kininogen 1) is a protein that is directly involved in the production of bradykinin, in the kallikrein-kinin system. Bradykinin is often called an inflammatory mediator, for its complex role in inflammatory systems.
Through different mechanisms, bradykinin can cause both vasoconstriction and vasoconstriction, acting as a sort of regulator.
With this, the effect of ~5-fold reduction in KNG1 expressions is unclear, however it is certain that doxycycline is having an impact on this system.

```{r plot-kng1, echo=FALSE}
# Find the specific summary data for Kng1
kng1_summary <- gene_summary_list[["Kng1"]]
kng1_summary$timepoint <- as.numeric(as.character(kng1_summary$timepoint))

ggplot(kng1_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(kng1_summary$timepoint)) +
  labs(title = "Kng1 Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## KNG1 IGV
![KNG1_IGV](figures/IGV_pics/Kng1_IGV.png)

```{r KNG1}
kng1_results <- stat_results_all %>% filter(gene == "Kng1")

print(kng1_results)
```

## SPN Downregulation
Spn (Sialophorin) is a protein on the surface of many cells, particularly immune cells, including T cells, monocytes, and granulocytes.
It is involved in immune signaling and inflammatory regulation.
A reduction in Spn expression may actually be pro-inflammatory because the protein regulates the behavior of immune cells, so lower levels can throw this system out of balance.
Additionally, lower Spn levels are correlated with reduced immune response, since mounting a proper response is reliant on signals from Spn.
While we noticed a steep decrease in the expression initially(almost 6-fold reduction within 48 hours), it seemed to recover significantly by the time we recorded the 96 hour timepoint, down to a little over a 2-fold reduction.
Again, we are not sure if this trend will continue, or if these expression changes will stick around long beyond the exposure to doxycycline.

```{r plot-spn, echo=FALSE}
# Find the specific summary data for Spn
spn_summary <- gene_summary_list[["Spn"]]
spn_summary$timepoint <- as.numeric(as.character(spn_summary$timepoint))
ggplot(spn_summary, aes(x = timepoint, y = mean, group = 1)) +
  geom_line() + geom_point() +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.2) +
  scale_x_continuous(breaks = unique(spn_summary$timepoint)) +
  labs(title = "Spn Expression Across Time", y = "Mean Count", x = "Time (hours)") +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
```

## SPN IGV
![SPN_IGV](figures/IGV_pics/Spn_IGV.png)
```{r Spn}
spn_results <- stat_results_all %>% filter(gene == "Spn")
knitr::kable(spn_results, digits = 3)
```

## Biological Interpretation of Modules

### Inflammatory/Stress Response
Genes: *Aoc3, Abcc2, Nlrp3, Kng1, Klf17, Spn*

## Inflammation Heatmap Visualization
```{r imflammatory-heatmap-visualization}
# Define the genes to include
genes_of_interest <- c("Aoc3", "Abcc2", "Nlrp3", "Kng1", "Klf17", "Spn")

# Filter the dataset for the genes of interest
infl_heatmap_matrix <- data_cleaned %>%
  filter(gene_name %in% genes_of_interest) %>%
  distinct(gene_name, .keep_all = TRUE) %>%
  select(gene_name, starts_with("WT")) %>%
  column_to_rownames("gene_name") %>%
  mutate(across(everything(), as.numeric)) %>%
  replace(is.na(.), 0) %>%  # Replace NAs with zeros
  as.matrix()

infl_heatmap_matrix[!is.finite(infl_heatmap_matrix)] <- 0

# Log2 transformation
infl_heatmap_matrix_log <- log2(infl_heatmap_matrix + 1)

infl_unclustered_heatmap <- pheatmap(
   infl_heatmap_matrix_log,
   cluster_rows = FALSE,
   cluster_cols = FALSE,
   scale = "row",
   fontsize = 10,
   fontsize_row = 8,
   fontsize_col = 8,
   angle_col = 45,
   main = "Log2 Expression Heatmap (Inflammation/Stress Genes)")

ggsave(filename = "figures/Graphs/Infl_heatmap.png", plot = infl_unclustered_heatmap, width = 8, height = 10)
```

## Conclusions
Our analyses reveal distinct biological modules triggered by doxycycline exposure:
an early inflammatory/stress response (potentially mitochondrial-related via Nlrp3) and a later metabolic/differentiation shift. 
Novel candidates like **Apol8** and **Klf17** emerge as key regulatory nodes for further experimental investigation.